In [70]:
import pandas as pd
import numpy as np

In [71]:
url = "https://raw.githubusercontent.com/dsoztorun/odev2_titanic_dsoztorun/refs/heads/main/Titanic_dataset.csv"
df_titanic = pd.read_csv(url)
df_titanic2 = df_titanic.copy()

In [72]:
df_titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


"Age" ve "Cabin" sütunlarında eksik değerler var. 
Modelleme değil analiz yaptığımızdan sadece eksik veriyi tespit ediyoruz.
Eksik veri silme ya da ortalama değer atama gibi işlemler yapmıyoruz.

In [73]:
print(df_titanic.isnull().sum().sum()) 
print(df_titanic.isnull().sum()) 

866
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


Toplam 866 boş hücre var. 
Sadece “Age” sütunundaki eksik değerleri silmek bile %19.86 veri kaybına neden olur.
.mean() ve .median() fonksiyonları null değerleri dışlayarak hesap yapar.

1.	Kazada ölenlerin yaş ortalamasını bulunuz

In [74]:
df_titanic[df_titanic["Survived"]==0]["Age"].mean()

np.float64(30.62617924528302)

Kazada ölenlerin yaş ortalaması ~30.63

2.	Kazada ölenlerin bilet fiyatlarının ortalamasını ve medyanını bulunuz

In [75]:
df_titanic[df_titanic["Survived"]==0]["Fare"].mean()

np.float64(22.117886885245902)

Kazada ölenlerin bilet fiyatlarının ortalaması ~22.12

In [76]:
df_titanic[df_titanic["Survived"]==0]["Fare"].median()

np.float64(10.5)

Kazada ölenlerin bilet fiyatlarının medyanı 10.5

3.	Kazada ölen erkeklerin yaş ortalamasını bulunuz

In [77]:
df_titanic[(df_titanic["Survived"]==0)&(df_titanic["Sex"]=="male")]["Age"].mean()

np.float64(31.618055555555557)

Kazada ölen erkeklerin yaş ortamalası ~31.62

4.	Kazada ölen kadınların yaş ortalamasını bulunuz

In [78]:
df_titanic[(df_titanic["Survived"]==0)&(df_titanic["Sex"]=="female")]["Age"].mean()

np.float64(25.046875)

Kazada ölen kadınların yaş ortalaması ~25

5.	Kazadan kurtulanların yaş ortalamasını bulunuz

In [79]:
df_titanic[df_titanic["Survived"]==1]["Age"].mean()

np.float64(28.343689655172415)

Kazadan kurtulanların yaş ortalaması ~28.34

6.	Kazadan kurtulanların bilet fiyatlarının ortalamasını bulunuz

In [80]:
df_titanic[df_titanic["Survived"]==1]["Fare"].mean()

np.float64(48.39540760233918)

Kazadan kurtulanların bilet fiyatlarının ortalaması ~48.4

7.	Kazadan kurtulan toplam kişi sayısını bulunuz…(tüyo: len(df_titanic[(df_titanic.Survived==1)] kodu ile de bu sayıya ulaşabilirsiniz.)

In [81]:
df_titanic[df_titanic["Survived"]==1]["Survived"].count()

np.int64(342)

Kazadan kurtulan toplam kişi sayısı 342

8.	10 yaşından küçüklerin bilet fiyatlarının medyan değerini bulunuz

In [82]:
df_titanic[df_titanic["Age"] < 10 ]["Fare"].median()

np.float64(27.0)

10 yaşından küçüklerin bilet fiyatlarının medyan değeri 27

9.	1.sınıf, 2.sınıf ve 3.sınıf bilet fiyatlarının ortalama ve medyanlarını karşılaştırınız. (Pclass değişkeni sınıfları barındırmaktadır.)

In [83]:
df_titanic[df_titanic["Pclass"]==1]["Fare"].mean()

np.float64(84.1546875)

In [84]:
df_titanic[df_titanic["Pclass"]==1]["Fare"].median()

np.float64(60.287499999999994)

In [85]:
df_titanic[df_titanic["Pclass"]==2]["Fare"].mean()

np.float64(20.662183152173913)

In [86]:
df_titanic[df_titanic["Pclass"]==2]["Fare"].median()

np.float64(14.25)

In [87]:
df_titanic[df_titanic["Pclass"]==3]["Fare"].mean()

np.float64(13.675550101832993)

In [88]:
df_titanic[df_titanic["Pclass"]==3]["Fare"].median()

np.float64(8.05)

In [89]:
Fare_Comparison_Dict = {"1.Class": [84.1546875,60.287499999999994],"2.Class":[20.662183152173913,14.25],"3.Class":[13.675550101832993,8.05]}

In [90]:
Fare_Comparison = pd.DataFrame(data = Fare_Comparison_Dict, index = ["Mean","Median"])

In [91]:
print(Fare_Comparison)

          1.Class    2.Class   3.Class
Mean    84.154687  20.662183  13.67555
Median  60.287500  14.250000   8.05000


1. sınıftan 3. sınıfa doğru bilet fiyatlarında ortalama ve medyanlar düşüş göstermektedir.

10.	Kazada ölen kadınların oranı ile erkeklerin oranını karşılaştırınız. (Örnek: erkekler için; ölen erkeklerin, erkek sayısına bölümü bu oranı vermektedir.

In [92]:
df_titanic[df_titanic["Sex"]=="female"]["Sex"].count()

np.int64(314)

In [93]:
df_titanic[(df_titanic["Survived"]==0)&(df_titanic["Sex"]=="female")]["Sex"].count()

np.int64(81)

In [94]:
df_titanic[df_titanic["Sex"]=="male"]["Sex"].count()

np.int64(577)

In [95]:
df_titanic[(df_titanic["Survived"]==0)&(df_titanic["Sex"]=="male")]["Sex"].count()

np.int64(468)

In [96]:
Cinsiyet_Olum_Dict = {"Kadinlar": [314, 81, 81/314*100], "Erkekler" : [577, 468, 468/577*100]}

In [97]:
Cinsiyet_Olum = pd.DataFrame(data=Cinsiyet_Olum_Dict, index = ["Toplam Kişi Sayisi", "Olen Sayisi", "Olum Orani(%)"], dtype=float)

In [98]:
print(Cinsiyet_Olum)

                      Kadinlar    Erkekler
Toplam Kişi Sayisi  314.000000  577.000000
Olen Sayisi          81.000000  468.000000
Olum Orani(%)        25.796178   81.109185


Kadınlarda ölüm oranı ~25.8 iken erkeklerde ~81.1'dir.